control-plane/agents/src/bin/core/tests/volume/format_options.rs

```diff
@@ -0,0 +1,164 @@
+    let reconcile_period = Duration::from_millis(3000);
+    let cluster = ClusterBuilder::builder()
+        .with_rest(true)
+        .with_io_engines(1)
+        .with_tmpfs_pool_ix(0, POOL_SIZE)
+        .with_csi(true, true)
+        .with_cache_period(&humantime::Duration::from(cache_period).to_string())
+        .with_reconcile_period(reconcile_period, reconcile_period)
+        .build()
+        .await
+        .expect("Failed to build cluster");
+
+    let api_client = cluster.rest_v00();
+    let volumes_api = api_client.volumes_api();
+    let vol_id = VolumeId::new();
+    let volume = volumes_api
+        .put_volume(
+            &vol_id,
+            models::CreateVolumeBody::new(
+                models::VolumePolicy::new(true),
+                1,
+                VOLUME_SIZE,
+                false,
+                false,
+            ),
+        )
+        .await
+        .expect("Failed to create volume");
+
+    let mut node = cluster
+        .csi_node_client(0)
+        .await
+        .expect("Failed to get node client");
+
+    let mut controller = cluster
+        .csi_controller_client()
+        .await
+        .expect("Failed to get controller client");
+
+    let publish_result = controller
+        .controller_publish_volume(&volume, &cluster.csi_node(0))
+        .await
+        .expect("Failed to publish volume");
+    let uri = publish_result
+        .publish_context
+        .get("uri")
+        .expect("Failed to get uri")
+        .to_string();
+
+    let _nvme_io_subsys = DeviceDisconnect(nvmeadm::NvmeTarget::try_from(uri).unwrap());
+
+    let mut volume_context = HashMap::new();
+    let flags = format!("-i {}", BYTES_PER_INODE);
+    volume_context.insert("formatOptions".to_string(), flags);
+
+    let vol = volumes_api
+        .get_volume(&vol_id)
+        .await
+        .expect("Cant get the volume");
+    node.node_stage_volume_fs(
+        &vol,
+        "ext4",
+        publish_result.publish_context.clone(),
+        volume_context,
+    )
+    .await
+    .expect("Failed to stage volume");
+    node.node_publish_volume_fs(&volume, "ext4", publish_result.publish_context)
+        .await
+        .expect("Failed to publish volume");
+
+    let path = format!("/var/tmp/target/mount/{}", volume.spec.uuid);
+    let expected_inodes = VOLUME_SIZE / BYTES_PER_INODE;
+
+    exhaust_inodes(path, expected_inodes).expect("Failed to exhaust inodes");
+
+    node.node_unpublish_volume(&volume)
+        .await
+        .expect("Failed to unpublish volume");
+    node.node_unstage_volume(&volume)
+        .await
+        .expect("Failed to unstage volume");
+    controller
+        .controller_unpublish_volume(&volume, &cluster.csi_node(0))
+        .await
+        .expect("Failed to publish volume");
+}
+
+/// Creates files enough to exhaust inode count of mount path present in csi-node-1 container.
+fn exhaust_inodes(path: String, inode_count: u64) -> Result<(), anyhow::Error> {
```

Comment:
we don't need to do this, just inspect output if tune2fs is what we expect
Also consider adding this test as a BDD test instead

Reply:
I tried libc statvfs module to get free inodes. However, Since path is mounted inside csi-node tune2fs/libc verification  was not working in test code. Same logic was giving correct inode count in csi-node stage/publish code. So thought of exhausting inodes.

Do you mean with more scenarios in BDD?

Reply:
You need to run the tune2fs command as a docker exec, like so: https://github.com/openebs/mayastor-control-plane/blob/51ba1c5491c67432db32da6b54d8e3dfc60a2811/tests/bdd/features/csi/node/test_node.py#L847